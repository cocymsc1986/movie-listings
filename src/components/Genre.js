import React from 'react';

const Genre = ({ name, id, onClick }) => {
    return (
        <div className="filters__genre" onClick={onClick}>
            <label htmlFor={`genre-${id}`}>{name}</label>
            <input type="checkbox" id={`genre-${id}`} value={id} />
        </div>
    );
};

export default Genre;