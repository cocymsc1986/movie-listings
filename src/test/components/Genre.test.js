import React from 'react';
import Genre  from '../../components/Genre';

describe('Genre component', () => {
    test('Should render', () => {
        const wrapper = shallow(<Genre name={'Name'} id={1} onClick={() => {}} />);

        expect(wrapper).toMatchSnapshot();
    });
});