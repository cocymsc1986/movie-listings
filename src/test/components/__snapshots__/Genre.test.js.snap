// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre component Should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Genre
    id={1}
    name="Name"
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="genre-1"
>
          Name
</label>,
        <input
          id="genre-1"
          type="checkbox"
          value={1}
/>,
      ],
      "className": "filters__genre",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name",
          "htmlFor": "genre-1",
        },
        "ref": null,
        "rendered": "Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "genre-1",
          "type": "checkbox",
          "value": 1,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="genre-1"
>
            Name
</label>,
          <input
            id="genre-1"
            type="checkbox"
            value={1}
/>,
        ],
        "className": "filters__genre",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
            "htmlFor": "genre-1",
          },
          "ref": null,
          "rendered": "Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "genre-1",
            "type": "checkbox",
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
